src/glue/jobs/json_to_parquet.py

```diff
@@ -332,13 +414,133 @@ def write_table_to_s3(
             format = "parquet",
             transformation_ctx="write_dynamic_frame")
 
+def count_records_for_event(
+        table: "pyspark.sql.dataframe.DataFrame",
+        event: str,
```

Comment:
Technically I can't. StrEnum was introduced in 3.11 and this runs in 3.10.

I've never used StrEnum as a type hint before. From the Enum tutorial it seems like it was designed to provide syntactic sugar for working with classes, rather than as a type hint. Is there a benefit to adding the type hint in addition to the docstring and the check we do inside the function itself?

Reply:
[I was more thinking of Enum in general](https://docs.python.org/3.10/library/enum.html#creating-an-enum) not specifically a `StrEnum` (I didn't even know that was a thing).

> Member values can be anything: [int](https://docs.python.org/3.10/library/functions.html#int), [str](https://docs.python.org/3.10/library/stdtypes.html#str), etc..

```
class EventType(Enum):
    """The event associated with a count."""
    
    
    READ = "READ"
    """This table has just now been read from the Glue table catalog
                and has not yet had any transformations done to it."""

    DROP_DUPLICATES = "DROP_DUPLICATES"
    """This table has just now had duplicate records dropped
                (see the function `drop_table_duplicates`)."""

    DROP_DELETED_SAMPLES = "DROP_DELETED_SAMPLES"
    """This table has just now had records which are
                present in its respective "Deleted" table dropped (see the function
                `drop_deleted_healthkit_data`)."""

    WRITE = "WRITE"
    """This table has just now been written to S3."""


def do_thing(event_type: EventType):
    print(event_type)


do_thing(EventType.READ)
do_thing(EventType.DROP_DUPLICATES)
do_thing(EventType.DROP_DELETED_SAMPLES)
do_thing(EventType.WRITE)
```

The point around this is that I don't need to know the string constant to use. I could check the docstring, but I could also use one of the available Enum values:

![image](https://github.com/Sage-Bionetworks/recover/assets/17128019/a3c23847-7edd-4f92-b411-d738055dcff3)

Reply:
That makes sense. It has some nifty features. I'll try it out.